### PR TITLE
MONGOID-3214 Ensure that update is not called multiple times for an embedded doc

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

File: spec/mongoid/persistable/savable_spec.rb [hunk 1]
```diff
@@ -83,6 +83,27 @@
             end
           end
         end
+
+        context 'when the embedded document is unchanged' do
+
+          before do
+            allow(Kangaroo.collection).to receive(:insert).once
```
Comment (simi) [inline spec/mongoid/persistable/savable_spec.rb]: I think it will be better to move expectations and stubs into the actual test block.

Comment (durran) [inline spec/mongoid/persistable/savable_spec.rb]: I agree here as well, since the expectation is the actual test here, not that the call to save is true.

Comment (estolfo) [inline spec/mongoid/persistable/savable_spec.rb]: ok, updated it.

**Comment by durran** [comment]
:+1:

File: spec/mongoid/persistable/savable_spec.rb [hunk 2]
```diff
@@ -83,6 +83,24 @@
...
         end
+
+        context 'when the embedded document is unchanged' do
+
+          let(:kangaroo) do
+            Kangaroo.new
+          end
+
+          after do
+            Kangaroo.destroy_all
```
Comment (simi) [inline spec/mongoid/persistable/savable_spec.rb]: :(

**Pull request merged**